lib/cluster.js

```diff
@@ -0,0 +1,211 @@
+					valObj.metricName = values[0].metricName;
+				}
+				// TODO how do we aggregate timestamps? Average? Omit?
+				result.values.push(valObj);
+			});
+			return result;
+		};
+	};
+
+	const SUM = AggregatorFactory(v => v.reduce((p, c) => p + c.value, 0));
+	const FIRST = AggregatorFactory(v => v[0].value);
+	const VOID = () => {};
+	const AVERAGE = AggregatorFactory(
+		v => v.reduce((p, c) => p + c.value, 0) / v.length
+	);
+	// const MIN = AggregatorFactory(v => v.reduce((p, c) => Math.min(p + c.value), 0));
+	// const MAX = AggregatorFactory(v => v.reduce((p, c) => Math.max(p + c.value), 0));
+
+	// Define merge strategies for all default metrics. User metrics may
+	// define an `aggregate` method, or will be summed by default.
+	// TODO These are here for the sake of isolation, but obviously would be
+	// nicer if they were defined in the metrics/ files.
+	// TODO Might be nice to have multiple aggregates for some, like the
+	// eventloop lag, where you might want MIN, MAX and AVERAGE.
+	const metricMergeFns = {
+		nodejs_active_handles_total: SUM,
+		nodejs_eventloop_lag_seconds: AVERAGE,
+		nodejs_heap_size_total_bytes: SUM,
+		nodejs_heap_size_used_bytes: SUM,
+		nodejs_external_memory_bytes: SUM,
+		nodejs_heap_space_size_total: SUM,
+		nodejs_heap_space_size_used: SUM,
+		nodejs_heap_space_size_available: SUM,
+		process_cpu_user_seconds_total: SUM,
+		process_cpu_system_seconds_total: SUM,
+		process_cpu_seconds_total: SUM,
+		process_max_fds: SUM,
+		process_open_fds: SUM,
+		nodejs_active_requests_total: SUM,
+		process_start_time_seconds: VOID, // don't think there's a sensible aggregate
+		nodejs_version_info: FIRST,
+		process_resident_memory_bytes: SUM,
+		process_virtual_memory_bytes: SUM,
+		process_heap_bytes: SUM
+	};
+	for (const metricName in metricMergeFns) {
+		const metric = Registry.globalRegistry.getSingleMetric(metricName);
+		// Not always defined: some are platform-specific.
+		if (metric) metric.aggregate = metricMergeFns[metricName];
+	}
+
+	/**
+	 * Gets aggregated metrics for all workers.
+	 * @param {Function} callback (err, metrics) => any
+	 * @return {undefined} undefined
+	 */
+	Registry.prototype.clusterMetrics = function(callback) {
+		const requestId = requestCtr++;
+
+		requests.set(requestId, {
+			responses: [],
+			pending: Object.keys(cluster.workers).length,
+			callback
+		});
+
+		const req = {
+			type: GET_METRICS_REQ,
+			requestId
+		};
+		for (const id in cluster.workers) cluster.workers[id].send(req);
+
+		// TODO set a timeout for worker responses.
+	};
+
+	/**
+	 * Creates a new Registry instance from an array of metrics that were
+	 * created by `registry.getMetricsAsJSON()`. Metrics are aggregated using
+	 * their `aggregate` method, or by summation if that is undefined.
+	 * @param {Array} metricsArr Array of metrics, each of which created by
+	 *     `registry.getMetricsAsJSON()`.
+	 * @return {Registry} aggregated registry.
+	 */
+	Registry.aggregate = function(metricsArr) {
```

Comment:
We have a `merge`, but that just includes different metrics and doesn't handle dupes?

If so, this function should be added next to merge, not live here